src/components/RangeSlider/components/DualThumb/DualThumb.tsx

```diff
@@ -0,0 +1,594 @@
+import * as React from 'react';
+import {autobind} from '@shopify/javascript-utilities/decorators';
+import {
+  addEventListener,
+  removeEventListener,
+} from '@shopify/javascript-utilities/events';
+import {classNames} from '@shopify/react-utilities/styles';
+import TextField from '../../../TextField';
+import {CSS_VAR_PREFIX} from '../../utilities';
+import {Props as RangeSliderProps} from '../../types';
+import Labelled from '../../../Labelled';
+
+import {Key} from '../../../../types';
+
+import * as styles from './DualThumb.scss';
+
+export interface State {
+  valueLower: number;
+  valueUpper: number;
+  trackWidth: number;
+}
+
+export interface Props extends RangeSliderProps {
+  value: [number, number];
+  id: string;
+  min: number;
+  max: number;
+  step: number;
+}
+
+const THUMB_SIZE = 24;
+const OUTPUT_TIP_SIZE = 8;
+
+export default class DualThumb extends React.Component<Props, State> {
+  state: State = {
+    valueLower: this.props.value[0],
+    valueUpper: this.props.value[1],
+    trackWidth: 0,
+  };
+
+  private track = React.createRef<HTMLDivElement>();
+  private thumbLower = React.createRef<HTMLButtonElement>();
+  private thumbUpper = React.createRef<HTMLButtonElement>();
+
+  componentDidMount() {
+    const {valueLower, valueUpper} = this.state;
+    const {step, min, max} = this.props;
+    const valueWithinBoundsLower = keepValueWithinBoundsLower(
+      roundToNearestStepValue(valueLower, step),
+      valueUpper,
+      min,
+      step,
+    );
+
+    const valueWithinBoundsUpper = keepValueWithinBoundsUpper(
+      roundToNearestStepValue(valueUpper, step),
+      valueLower,
+      max,
+      step,
+    );
+
+    this.setState({
+      valueLower: valueWithinBoundsLower,
+      valueUpper: valueWithinBoundsUpper,
+    });
+
+    if (this.track.current) {
+      this.setState({
+        trackWidth: this.track.current.getBoundingClientRect().width,
```

Comment:
track width also needs to be updated when its container resizes, with either a resize observer or a window resize event

Reply:
For traceability: https://github.com/Shopify/polaris-react/issues/840